### PR TITLE
Add oVirt storage management module

### PR DESCRIPTION
**Pull request by rgroten**
##### ISSUE TYPE
- New Module Pull Request
##### COMPONENT NAME

ovirt_storage
##### ANSIBLE VERSION

```
ansible 2.2.0 (devel b361bf90d7) last updated 2016/06/27 14:57:32 (GMT -600)
  lib/ansible/modules/core: (detached HEAD 3c6f2c2db1) last updated 2016/06/27 14:58:37 (GMT -600)
  lib/ansible/modules/extras: (detached HEAD 1c36665545) last updated 2016/06/27 14:59:31 (GMT -600)
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
```
##### SUMMARY

This module provides the ability to perform storage related operations, such as create, attach, detach, or delete disks in the oVirt or RHEV virtualization products.

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks @rgroten for this new module. When this module receives 'shipit' comments from two community members and any 'needs_revision' comments have been resolved, we will mark for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by rgroten** [comment]
I'm not familiar with the community_review process, is there something I need to do to get others to get this reviewed?  Or is it more volunteer based?

File: cloud/misc/ovirt_storage.py [hunk 1]
```diff
@@ -0,0 +1,432 @@
...
+        description:
+            - User with access to perform disk operations in oVirt.
+        required: true
+    password:
+        description:
+            - Password for I(user).
+        required: true
+    url:
+        description:
+            - URL to oVirt API (U(https://<ovirt-engine>/api)).
```
Comment (jhernand) [inline cloud/misc/ovirt_storage.py]: The path of the oVirt API should be `/ovirt-engine/api`. The old `/api` has been deprecated and will be removed in version 4.1 of oVirt.

File: cloud/misc/ovirt_storage.py [hunk 2]
```diff
@@ -0,0 +1,432 @@
...
+        disks = vm.disks.list(alias=disk_alias)
+        if disks:
+            return 0
+        if not MODULE_CHECK_MODE:
+            size = int(disk_size_gb) * 1024 * 1024 * 1024
+            disk_params = params.Disk()
+            disk_params.set_wipe_after_delete(True)
+            disk_params.set_active(True)
+            disk_params.set_alias(disk_alias)
+            disk_params.set_size(size)
```
Comment (jhernand) [inline cloud/misc/ovirt_storage.py]: The `size` attribute of disks has been deprecated, and replaced by `provisioned_size`. It is just a name change, consider using the new name.

**Comment by rgroten** [comment]
Thanks @jhernand for the notes, I made the changes you recommended.

**Comment by gregdek** [comment]
Thanks @rgroten for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by jhernand** [comment]
Looks good!

File: cloud/misc/ovirt_storage.py [hunk 3]
```diff
@@ -0,0 +1,432 @@
...
+        description:
+            - If C(yes), disk will be activated when created or attached to a VM.
+            - If C(no), disk will be attached but not activated.
+            - Used when C(state) is C(present) or C(attached).
+        default: True
+        type: bool
+'''
+
+EXAMPLES = '''
+# Create a new 5GB Pool Disk and attach/activate it to a VM
```
Comment (rollandf) [inline cloud/misc/ovirt_storage.py]: What do you mean by "Pool" ?

Comment (rgroten) [inline cloud/misc/ovirt_storage.py]: @rollandf You're right that's confusing, it probably should be renamed to Image, I will make that change now.

**Comment by rgroten** [comment]
@gregdek I'm not sure, but it seems like the build is failing on something other than my module?

**Comment by rollandf** [comment]
shipit

**Comment by rgroten** [comment]
ready_for_review

(I'm not sure how to fix the failed build, but the module passed build earlier and only semantics changes have been made since)

**Comment by gregdek** [comment]
Thanks @rgroten for this new module. When this module receives 'shipit' comments from two community members and any 'needs_revision' comments have been resolved, we will mark for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
Thanks @rgroten for this new module. When this module receives 'shipit' comments from two community members and any 'needs_revision' comments have been resolved, we will mark for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by rgroten** [comment]
#2836 covers everything this module does so this is not necessary anymore.

**Pull request closed without merging**